crates/node/src/providers/robust_ecdsa.rs

```diff
@@ -0,0 +1,284 @@
+pub mod presign;
+mod sign;
+
+use mpc_contract::primitives::key_state::KeyEventId;
+pub use presign::PresignatureStorage;
+use std::collections::HashMap;
+
+use crate::config::{ConfigFile, MpcConfig, ParticipantsConfig};
+use crate::db::SecretDB;
+use crate::network::{MeshNetworkClient, NetworkTaskChannel};
+use crate::primitives::{MpcTaskId, UniqueId};
+use crate::providers::{EcdsaSignatureProvider, SignatureProvider};
+use crate::storage::SignRequestStorage;
+use crate::tracking;
+
+use crate::types::SignatureId;
+use borsh::{BorshDeserialize, BorshSerialize};
+use mpc_contract::primitives::domain::DomainId;
+use near_time::Clock;
+use std::sync::Arc;
+use threshold_signatures::ecdsa::KeygenOutput;
+use threshold_signatures::ecdsa::Signature;
+use threshold_signatures::frost_secp256k1::keys::SigningShare;
+use threshold_signatures::frost_secp256k1::VerifyingKey;
+
+pub struct RobustEcdsaSignatureProvider {
+    config: Arc<ConfigFile>,
+    mpc_config: Arc<MpcConfig>,
+    client: Arc<MeshNetworkClient>,
+    sign_request_store: Arc<SignRequestStorage>,
+    per_domain_data: HashMap<DomainId, PerDomainData>,
+}
+
+#[derive(Clone)]
+pub(super) struct PerDomainData {
+    pub keyshare: KeygenOutput,
+    pub presignature_store: Arc<PresignatureStorage>,
+}
+
+#[derive(Debug, Clone, Eq, Ord, PartialEq, PartialOrd, derive_more::From, derive_more::Into)]
+pub struct EcdsaMessageHash([u8; 32]);
+
+impl EcdsaMessageHash {
+    pub fn as_bytes(&self) -> [u8; 32] {
+        self.0
+    }
```

Comment:
Two notes on this method:

1. (non-blocking suggestion) - I think it would be more idiomatic to use deref instead of an ad-hoc conversion method since this is a simple wrapper type. Consider deriving `derive_more::Deref`.
2. (blocking) - If we keep this method it should be renamed to `to_bytes` since it copies the underlying data as per the naming conventions in https://rust-lang.github.io/api-guidelines/naming.html?highlight=convers#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv. It should also take ownership of `self`, and we could derive `Copy` on the wrapper as well.